mlflow/cli.py

```diff
@@ -519,11 +520,20 @@ def server(
     " are not specified, data is removed for all runs in the `deleted`"
     " lifecycle stage.",
 )
-def gc(older_than, backend_store_uri, run_ids):
+@click.option(
+    "--experiment-ids",
```

Comment:
I'm wondering what should happen when both `--experiment-ids` and `--older-than` are specified:

```
mlflow gc --experiment-ids 123 --older-than 1d
```

Should this command behave like "delete experiment 123 if it's older than 1d, otherwise throw"?


@dbczumar Thoughts?

Reply:
@harupy I have taken the reference of a similar scenario w.r.t runs, which is throwing an Exception. Do I need to modify both runs and experiments to produce warning messages instead? If so, how warning messages are tested?

Reply:
Here's an example of using `with pytest.warns()` 
https://github.com/mlflow/mlflow/blob/9f3d3afea71c47657b47ec72def36a04bbb13df3/tests/models/test_model.py#L152-L157

Reply:
@BenWilson2 Thanks for sharing an example. What should be the type of warning?

Reply:
Typically the default UserWarning, when applied to "dear user, I am warning you because of something that you told me to do that I cannot do, sincerely, your computer" is pretty sufficient.

Reply:
@BenWilson2  I like the humor. Can I keep the warning message in similar lines, if it's not offensive?

Reply:
@shaikmoeed I think that preserving the same exception behavior used for runs is the best approach here. @BenWilson2 let me know if you have concerns about using an exception rather than a warning. My reasoning here is that the user is asking `mlflow gc` to do something invalid / contradictory: "Delete a particular experiment that is less than X days old and ensure that all deleted experiments are more than X days old".

Reply:
@BenWilson2 @dbczumar Thanks for the comments. I'd vote for preserving the same exception behavior as Corey suggested.